### ISSUE TITLE
Standartize the backup format

### ISSUE THREAD

**Issue by StoyanDimitrov** [andOTP/andOTP#360]
There is an [ongoing](https://github.com/helloworld1/FreeOTPPlus/issues/30) [initiative](https://github.com/Authenticator-Extension/Authenticator/issues/282) to standardize the backup format of several other OTP authenticators.

I suggest andOTP joins that movement.

**Comment by RichyHBM** [comment]
I support the idea, but has there been any decided standard on what this format would be?

**Comment by flocke** [comment]
I totally agree with @RichyHBM and fully support a standardized backup format. I have a few free days next week and will try to look into the discussion you linked and maybe particiapte. As soon as a standard has been decided on I will start on implementing it in andOTP.

**Comment by captainju** [comment]
> I support the idea, but has there been any decided standard on what this format would be?

Yes https://github.com/Authenticator-Extension/Authenticator/wiki/Standard-OTP-Backup-Format-Devloper-Info

**Comment by RichyHBM** [comment]
The problem with that format is it wouldnt allow us to save any image/icon information with entries, maybe we can add a custom &image=... argument?

**Comment by captainju** [comment]
Yes the custom arg should be enough

**Comment by fortran77** [comment]
Just wanted to add a couple of comments.

Just the ability to exchange the essential TOTP information, without icons, will be very helpful to users. It's just one otpauth URI per line, so implementation can be quick.

Later the format can be further extended by agreement between developers to add:

- Icons.
- Encryption.
- An end marker so file truncation can be detected
- A comment syntax in case one wishes to manually annotate an exported file.
- Checksums to detect corrruption.
